### PR TITLE
fix: apt.deb_index should work without a lock

### PR DESCRIPTION
**Pull request by thesayyn**
Fixes the issue where if no `lock` file is passed to `apt.deb_index` it fails horribly.  This change fixes the issue.
Sorry for the big diff because i had to refactor a lot to make it work with bzlmod!

### CODE REVIEW

File: examples/debian_shared_dependencies/BUILD.bazel [hunk 1]
```diff
@@ -92,7 +92,7 @@ oci_image(
     }),
 )
 
-oci_tarball(
+oci_load(
```
Comment (alexeagle) [inline examples/debian_shared_dependencies/BUILD.bazel:95]: nit: is this used anywhere?

File: e2e/smoke/MODULE.bazel [hunk 2]
```diff
@@ -1,22 +1,26 @@
...
 
 apt = use_extension("@rules_distroless//apt:extensions.bzl", "apt")
-apt.deb_index(
+apt.install(
     name = "bullseye",
     lock = ":bullseye.lock.json",
     manifest = ":bullseye.yaml",
 )
+apt.install(
+    name = "bullseye_nolock",
```
Comment (alexeagle) [inline e2e/smoke/MODULE.bazel:20]: nit: maybe comment about why this is needed?

File: apt/private/resolve.bzl [hunk 3]
```diff
@@ -61,8 +19,11 @@ def _parse_manifest(rctx):
 
     return json.decode(result.stdout if result.stdout != "null" else "{}")
 
-def _deb_resolve_impl(rctx):
-    manifest = _parse_manifest(rctx)
+# This function is shared between BZLMOD and WORKSPACE implementations.
+# INTERNAL: DON NOT DEPEND!
```
Comment (alexeagle) [inline apt/private/resolve.bzl:23]: nit: spelling

File: apt/private/deb_import.bzl [hunk 4]
```diff
@@ -0,0 +1,29 @@
+"deb_import"
+
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+
+# BUILD.bazel template
```
Comment (alexeagle) [inline apt/private/deb_import.bzl:5]: nit: maybe it should go in `BUILD.deb` following typical convention?

**Review by alexeagle — APPROVED** [review]

File: examples/debian_shared_dependencies/BUILD.bazel (hunk 1 above)
Comment (thesayyn) [reply examples/debian_shared_dependencies/BUILD.bazel:95]: not by our tests no, but there for the completeness of the example

File: e2e/smoke/MODULE.bazel (hunk 2 above)
Comment (thesayyn) [reply e2e/smoke/MODULE.bazel:20]: Sure, the doc here explains why. i will leave comment there for visibility. https://github.com/GoogleContainerTools/rules_distroless/blob/main/docs/apt.md#lockfile-generation

File: apt/private/deb_import.bzl (hunk 4 above)
Comment (thesayyn) [reply apt/private/deb_import.bzl:5]: Hmm, it could go into a file and passed to http_archive build_file attribute. I'll reconsider it in the future.

**Pull request merged**